game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/fire/FireRoundStepsFactoryTest.java

```diff
@@ -0,0 +1,131 @@
+                    (arg1) ->
+                        FiringGroup.groupBySuicideOnHit(
+                            "army", List.of(givenAnyUnit()), List.of(mock(Unit.class))))
+                .casualtySelector((arg1, arg2) -> new CasualtyDetails())
+                .side(OFFENSE)
+                .battleState(
+                    FakeBattleState.givenBattleStateBuilder()
+                        .attacker(attacker)
+                        .defender(defender)
+                        .build())
+                .build()
+                .createSteps());
+
+    assertThat(names, is(generalFightStepStrings(attacker, defender, "army")));
+  }
+
+  private List<String> getStepNames(final List<BattleStep> steps) {
+    return steps.stream().flatMap(step -> step.getNames().stream()).collect(Collectors.toList());
+  }
+
+  @Test
+  @DisplayName("With two groups, there should be two sets of fire round strings")
+  void getNamesWithTwoFiringGroups() {
+    final GamePlayer attacker = mock(GamePlayer.class);
+    when(attacker.getName()).thenReturn("attacker");
+    final GamePlayer defender = mock(GamePlayer.class);
+    when(defender.getName()).thenReturn("defender");
+    final List<String> names =
+        getStepNames(
+            FireRoundStepsFactory.builder()
+                .battleActions(mock(BattleActions.class))
+                .returnFire(MustFightBattle.ReturnFire.ALL)
+                .diceRoller((arg1, arg2) -> new DiceRoll())
+                .firingGroupSplitter(
+                    (arg1) ->
+                        List.of(
+                            FiringGroup.groupBySuicideOnHit(
+                                    "army", List.of(givenAnyUnit()), List.of(mock(Unit.class)))
+                                .get(0),
+                            FiringGroup.groupBySuicideOnHit(
+                                    "spies", List.of(givenAnyUnit()), List.of(mock(Unit.class)))
+                                .get(0)))
+                .casualtySelector((arg1, arg2) -> new CasualtyDetails())
+                .side(OFFENSE)
+                .battleState(
+                    FakeBattleState.givenBattleStateBuilder()
+                        .attacker(attacker)
+                        .defender(defender)
+                        .build())
+                .build()
+                .createSteps());
+
+    assertThat(
+        names,
+        is(
+            mergeSteps(
+                generalFightStepStrings(attacker, defender, "army"),
+                generalFightStepStrings(attacker, defender, "spies"))));
+  }
+
+  @Test
+  @DisplayName("With the default name UNITS, the step names should be readable")
+  void getNamesWithDefaultUnitName() {
+    final GamePlayer attacker = mock(GamePlayer.class);
+    when(attacker.getName()).thenReturn("attacker");
+    final GamePlayer defender = mock(GamePlayer.class);
+    when(defender.getName()).thenReturn("defender");
+    final List<String> names =
```

Comment:
nit, I notice there is no new line between the test setup and the invocation of the unit under test. Having the setup, invocation of the test unit, and the assertions be separated by a new line helps delimit them and helps the reader easily see what is being tested vs the setup. (Arrange, Act, Assert (AAA) testing pattern). Any objection to inserting such new lines?

For example, 
```
testMethod() {
    setupVar = setup...
    player = ... setup
    otherState = ... setup

    resultValue = methodUnderTest.invoke(player, otherState ...)

    assertions(....) 
}
```

### Side note on AAA

I've even some styles where developers use boiler plate templates with comment blocks to delimit these sections, eg:
```
testMethod() {
   // arrange

   // act

   // assert
}
```

I"ve also seen those comments be called "given, when, then". IMO the comment blocks add clutter and the test method should be simple enough that simple new lines are delimiter enough to mark the different parts of a test. The latter should be possible almost all of the time, particularly if any complicated arrangements are extracted to helper methods that give a high level overview of what we are setting up with parameters for the key values that would control the result value of the unit under test.